### PR TITLE
Allow overriding the constructor

### PR DESCRIPTION
**Pull request by greg0ire**
Follow up of https://github.com/doctrine/dbal/pull/5036, prompted by https://github.com/doctrine/DoctrineBundle/issues/1246#issuecomment-1190884340

### CODE REVIEW

File: src/Types/Type.php [hunk 1]
```diff
@@ -48,13 +48,6 @@ abstract class Type
 
     private static ?TypeRegistry $typeRegistry = null;
 
-    /**
-     * @internal Do not instantiate directly - use {@see Type::addType()} method instead.
```
Comment (stof) [inline src/Types/Type.php:52]: A type registered through `addType` is still required to have no arguments in the constructor. We still need a check for that.

Comment (morozov) [reply src/Types/Type.php:52]: It looks like we need to deprecate the static `Type` methods that accept `string $className` in favor of `TypeRegistry` and the corresponding counterparts which accept `Type $type`.

Only after the deprecated methods have been removed, this constraint could be dropped.

Comment (greg0ire) [reply src/Types/Type.php:52]: See https://github.com/doctrine/dbal/pull/5530

**Comment by morozov** [comment]
Given that there are unresolved dependencies not being actively worked on (see https://github.com/doctrine/dbal/pull/5530#discussion_r932462816), I propose that we close this pull request for now.

**Pull request closed without merging**